Ensure enough BNB in account for withdraw

+1 

I think if the the telegram would send bnb as well as rune, it would solve the issue of 

`8-es2015.83359b12f00dfdfbb407.js:1 error making transfer:  Error: insufficient fund. you got 100000000000RUNE-67C, but 37500BNB fee needed.
    at main-es2015.f32b5e25744d56ffd8c7.js:1
    at u.invoke (polyfills-es2015.73c93b7576c046948b8f.js:1)
    at Object.onInvoke (main-es2015.f32b5e25744d56ffd8c7.js:1)
    at u.invoke (polyfills-es2015.73c93b7576c046948b8f.js:1)
    at a.run (polyfills-es2015.73c93b7576c046948b8f.js:1)
    at polyfills-es2015.73c93b7576c046948b8f.js:1
    at u.invokeTask (polyfills-es2015.73c93b7576c046948b8f.js:1)
    at Object.onInvokeTask (main-es2015.f32b5e25744d56ffd8c7.js:1)
    at u.invokeTask (polyfills-es2015.73c93b7576c046948b8f.js:1)
    at a.runTask (polyfills-es2015.73c93b7576c046948b8f.js:1)`

--- a/.gitignore
+++ b/.gitignore
@@ -46,4 +46,4 @@ testem.log
 Thumbs.db
 
 .env
-/src/environments/environment.ts
+src/environments/*

--- a/.gitignore
+++ b/.gitignore
@@ -46,3 +46,4 @@ testem.log
 Thumbs.db
 
 .env
+/src/environments/environment.ts

--- a/src/app/_components/user-settings/user-settings-dialog/confim-send/confim-send.component.ts
+++ b/src/app/_components/user-settings/user-settings-dialog/confim-send/confim-send.component.ts
@@ -1,5 +1,5 @@
 import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
-import { assetAmount, assetToBase } from '@thorchain/asgardex-util';
+import { assetAmount, assetToBase, baseAmount } from '@thorchain/asgardex-util';
 import { Subscription } from 'rxjs';
 import { AssetAndBalance } from 'src/app/_classes/asset-and-balance';
 import { User } from 'src/app/_classes/user';
@@ -85,9 +85,12 @@ export class ConfimSendComponent implements OnInit, OnDestroy {
         try {
 
           const feeRates = await bitcoinClient.getFeeRates();
+          const fees = await bitcoinClient.getFees();
+          const toBase = assetToBase(assetAmount(this.amount));
+          const amount = toBase.amount().minus(fees.average.amount());
 
           const hash = await bitcoinClient.transfer({
-            amount: assetToBase(assetAmount(this.amount)),
+            amount: baseAmount(amount),
             recipient: this.recipientAddress,
             feeRate: feeRates.average
           });

--- a/src/app/deposit/deposit.component.ts
+++ b/src/app/deposit/deposit.component.ts
@@ -86,6 +86,7 @@ export class DepositComponent implements OnInit, OnDestroy {
 
   user: User;
   coinGeckoList: CGCoinListItem[];
+  insufficientBnb: boolean;
   subs: Subscription[];
 
   constructor(
@@ -100,12 +101,13 @@ export class DepositComponent implements OnInit, OnDestroy {
 
     const balances$ = this.userService.userBalances$.subscribe(
       (balances) => {
-        console.log('balances are: ', balances);
         this.balances = balances;
-        console.log('this rune is: ', this.rune);
         this.runeBalance = this.userService.findBalance(this.balances, this.rune);
-        console.log('rune balance is: ', this.runeBalance);
         this.assetBalance = this.userService.findBalance(this.balances, this.asset);
+
+        // allows us to ensure enough bnb balance
+        const bnbBalance = this.userService.findBalance(this.balances, new Asset('BNB.BNB'));
+        this.insufficientBnb = bnbBalance < 0.000375;
       }
     );
 
@@ -173,7 +175,7 @@ export class DepositComponent implements OnInit, OnDestroy {
 
   formDisabled(): boolean {
 
-    return !this.balances || !this.runeAmount || !this.assetAmount
+    return !this.balances || !this.runeAmount || !this.assetAmount || this.insufficientBnb
     || (this.balances
       && (this.runeAmount > this.runeBalance || this.assetAmount > this.userService.maximumSpendableBalance(this.asset, this.assetBalance))
     );
@@ -188,6 +190,8 @@ export class DepositComponent implements OnInit, OnDestroy {
     } else if (this.balances && (this.runeAmount > this.runeBalance
       || this.assetAmount > this.userService.maximumSpendableBalance(this.asset, this.assetBalance))) {
       return 'Insufficient balance';
+    } else if (this.insufficientBnb) {
+      return 'Insufficient BNB for Fee';
     } else if (this.balances && this.runeAmount && this.assetAmount
       && (this.runeAmount <= this.runeBalance) && (this.assetAmount <= this.assetBalance)) {
       return 'Deposit';

--- a/src/app/withdraw/withdraw.component.html
+++ b/src/app/withdraw/withdraw.component.html
@@ -85,12 +85,13 @@
       class="main-action-btn"
       mat-flat-button
       color="primary"
-      [disabled]="!user || !removeAssetAmount || (removeAssetAmount && removeAssetAmount <= 0 || remainingTime)"
+      [disabled]="!user || !removeAssetAmount || insufficientBnb || (removeAssetAmount && removeAssetAmount <= 0 || remainingTime)"
     >
       <span *ngIf="!user">Please connect wallet</span>
       <span *ngIf="user && remainingTime">Withdraw Enabled in {{remainingTime}}</span>
-      <span *ngIf="user && !remainingTime && !removeAssetAmount || (removeAssetAmount && removeAssetAmount <= 0)">Enter an amount</span>
-      <span *ngIf="user && !remainingTime && removeAssetAmount && removeAssetAmount > 0">Withdraw</span>
+      <span *ngIf="user && !remainingTime && insufficientBnb">Insufficient BNB for Fee</span>
+      <span *ngIf="user && !remainingTime && !removeAssetAmount && !insufficientBnb || (removeAssetAmount && removeAssetAmount <= 0)">Enter an amount</span>
+      <span *ngIf="user && !remainingTime && removeAssetAmount && !insufficientBnb && removeAssetAmount > 0">Withdraw</span>
     </button>
 
   </div>

--- a/src/app/withdraw/withdraw.component.ts
+++ b/src/app/withdraw/withdraw.component.ts
@@ -50,6 +50,8 @@ export class WithdrawComponent implements OnInit {
   runeBasePrice: number;
   assetBasePrice: number;
 
+  insufficientBnb: boolean;
+
   constructor(
     private dialog: MatDialog,
     private route: ActivatedRoute,
@@ -72,12 +74,20 @@ export class WithdrawComponent implements OnInit {
       }
     );
 
+    const balances$ = this.userService.userBalances$.subscribe(
+      (balances) => {
+        // allows us to ensure enough bnb balance
+        const bnbBalance = this.userService.findBalance(balances, new Asset('BNB.BNB'));
+        this.insufficientBnb = bnbBalance < 0.000375;
+      }
+    );
+
     const lastBlock$ = this.lastBlockService.lastBlock$.subscribe( (block) => {
       this.lastBlock = block;
       this.checkCooldown();
     });
 
-    this.subs = [user$, lastBlock$];
+    this.subs = [user$, lastBlock$, balances$];
   }
 
   ngOnInit(): void {

